### DISCUSSION TITLE
Re-using best equation strings

### DISCUSSION THREAD

**Discussion by KevinG1002** [MilesCranmer/PySR#707] [Q&A]
Hello, 

New user here (SymbolicRegression.jl) and I absolutely love it! 

I was wondering if anyone knew a way to turn the equation strings that are obtained after n-iterations into a callable function so that, for example: 

`returned_equ = (((x₅ * -50) * x₁) + -2)` could be easily turned into a callable like `f = turn_to_callable(equation_str)` with `f(x_1, x_2, ..., x_n) = ....` yielding the result of the computation in the equation string (a bit similar to what `Polynomials.jl` does). I am essentially looking for something that could be easily applied to any new vector `[x_1, x_2, ..., x_n]` without going through `predict(mach, x_test)`. 

Perhaps, to simplify things, how does `predict(mach, x_test)` work exactly?

**Comment by MilesCranmer** [comment]
Glad to hear you are liking it! :)

If you run `report(mach)` you will get a NamedTuple containing an element `equations`. You can use this to access the callable functions for each equation:

```julia
using MLJ, SymbolicRegression

# ...

fit!(mach)

equations = report(mach).equations
```

This is a vector of callable expression objects. You can call these with:

```julia
eq = equations[end]  # Most accurate expression is last one
# Note that the default selection is `report(mach).best_idx`

formatted_X = MLJ.matrix(X)'  # Convert to matrix, and transpose to column-major
options = SymbolicRegression.MLJInterfaceModule.get_options(model)  # Equivalent of `Options(...)`

eq(formatted_X, options)
```

This will evaluate the resultant equation on `X`.

Sorry that you need to call `get_options` manually right now which is admittedly not great. However, once https://github.com/MilesCranmer/SymbolicRegression.jl/pull/326 finally merges, it will be much easier to use this – you won't even need to pass the `options` object anymore as it will be embedded in the expression.
